src/MongoDB.Driver.Core/Core/Misc/DnsClientWrapper.cs

```diff
@@ -38,6 +43,30 @@ public DnsClientWrapper()
         public LookupClient LookupClient => _lookupClient;
 
         // public methods
+        public string GetCanonicalizedHostName(string hostNameOrAddress)
+        {
+#if NETSTANDARD1_5 || NETSTANDARD1_6
+            var hostEntry = Dns.GetHostEntryAsync(hostNameOrAddress).GetAwaiter().GetResult();
+#else
+            var hostEntry = Dns.GetHostEntry(hostNameOrAddress);
```

Comment:
I'm not sure about this line, since internal steps in this method are not the same for `ip address`es comparing them with Python implementation. In input argument is `ip address`, we call `getnameinfo(ipAddress)` to receive hostName info which in turn will be used as input parameter `GetAddrInfo`. So, `getnameinfo` is the step which is not presented in the Python code